### PR TITLE
feat(terraform): add CKV NCP rules about access control group outbound rule.

### PR DESCRIPTION
**Pull request by Floodnut**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description
Add terraform scan rules about provider Naver Cloud Platform.

CKV_NCP_3 is associated with Access Control Group Outbound Rule.
This rule suggests no access control groups allow outbound from 0.0.0.0:0 to port.

### Description
- https://registry.terraform.io/providers/NaverCloudPlatform/ncloud/latest/docs/resources/access_control_group_rule

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/terraform/checks/resource/ncp/AccessControlGroupOutboundRule.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+
+class AccessControlGroupOutboundRule(BaseResourceCheck):
+    def __init__(self):
+        name = "An outbound security group rule allows traffic to /0"
+        id = "CKV_NCP_3"
+        supported_resources = ['ncloud_access_control_group_rule']
+
+        categories = [CheckCategories.NETWORKING]
+        guideline = "You should restrict access to IP addresses or ranges that are explicitly required where possible."
```
Comment (gruebel) [inline checkov/terraform/checks/resource/ncp/AccessControlGroupOutboundRule.py:12]:
```suggestion
```
guidelines for included checks are added separately by us, so no need to do it here. just add some explanation in the description of the PR 🙂  and we will use it when we create the docs for it

File: checkov/terraform/checks/resource/ncp/AccessControlGroupOutboundRule.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+from checkov.terraform.checks.resource.base_resource_check import BaseResourceCheck
+from checkov.common.models.enums import CheckResult, CheckCategories
+
+
+class AccessControlGroupOutboundRule(BaseResourceCheck):
+    def __init__(self):
+        name = "An outbound security group rule allows traffic to /0"
```
Comment (gruebel) [inline checkov/terraform/checks/resource/ncp/AccessControlGroupOutboundRule.py:7]:
```suggestion
        name = "Ensure no security group rules allow outbound traffic to 0.0.0.0/0"
```
typically we start our check names with `Ensure ...`

File: checkov/terraform/checks/resource/ncp/AccessControlGroupOutboundRule.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        categories = [CheckCategories.NETWORKING]
+        guideline = "You should restrict access to IP addresses or ranges that are explicitly required where possible."
+        super().__init__(name=name, id=id, categories=categories,
+                         supported_resources=supported_resources, guideline=guideline)
+
+    def scan_resource_conf(self, conf):
+        if 'outbound' in conf.keys():
+            for inbound in conf['outbound']:
+                ip = inbound.get('ip_block', '0.0.0.0/0')[0]
+                if ip == '0.0.0.0/0' or ip == '::/0':
```
Comment (gruebel) [inline checkov/terraform/checks/resource/ncp/AccessControlGroupOutboundRule.py:20]:
```suggestion
            for outbound in conf['outbound']:
                ip = outbound.get('ip_block')
                if ip == ['0.0.0.0/0'] or ip == ['::/0']:
```
you probably meant `outbound` here 😄 
this way is a bit safer, if `ip_block` is a value, which can't identify.

File: tests/terraform/checks/resource/ncp/example_LBTargetGroupHTTPS/main.tf [hunk 4]
```diff
@@ -0,0 +1,54 @@
+resource "ncloud_lb_target_group" "pass" {
```
Comment (gruebel) [inline tests/terraform/checks/resource/ncp/example_LBTargetGroupHTTPS/main.tf:1]: I think this whole file doesn't belong to this check, right?

**Review by gruebel — COMMENTED** [review]
nice work, left some comments

File: tests/terraform/checks/resource/ncp/example_LBTargetGroupHTTPS/main.tf (hunk 4 above)
Comment (pj991207) [reply tests/terraform/checks/resource/ncp/example_LBTargetGroupHTTPS/main.tf:1]: Sorry. I quickly delete it

**Review by gruebel — APPROVED** [review]
thanks, looks great 🍻

**Review by ayajbara — APPROVED** [review]

File: checkov/terraform/checks/resource/ncp/AccessControlGroupOutboundRule.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        id = "CKV_NCP_3"
+        supported_resources = ['ncloud_access_control_group_rule']
+
+        categories = [CheckCategories.NETWORKING]
+        super().__init__(name=name, id=id, categories=categories,
+                         supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf):
+        if 'outbound' in conf.keys():
+            for outbound in conf['outbound']:
```
Comment (Eliran-Turgeman) [inline checkov/terraform/checks/resource/ncp/AccessControlGroupOutboundRule.py:17]:
```suggestion
        for outbound in conf.get('outbound', []):
```

File: checkov/terraform/checks/resource/ncp/AccessControlGroupOutboundRule.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+from checkov.common.models.enums import CheckResult, CheckCategories
+
+
+class AccessControlGroupOutboundRule(BaseResourceCheck):
+    def __init__(self):
+        name = "Ensure no security group rules allow outbound traffic to 0.0.0.0/0"
+        id = "CKV_NCP_3"
+        supported_resources = ['ncloud_access_control_group_rule']
+
+        categories = [CheckCategories.NETWORKING]
```
Comment (Eliran-Turgeman) [inline checkov/terraform/checks/resource/ncp/AccessControlGroupOutboundRule.py:11]:
```suggestion
        supported_resources = ('ncloud_access_control_group_rule',)

        categories = (CheckCategories.NETWORKING,)
```

**Review by Eliran-Turgeman — APPROVED** [review]
Looking great! 
Added a few suggestions

**Comment by Floodnut** [comment]
I adjust additional suggestion!

**Pull request merged**